packages/grid/src/vaadin-grid-sort-mixin.js

```diff
@@ -126,22 +139,22 @@ export const SortMixin = (superClass) =>
     }
 
     /** @private */
-    __updateSorter(sorter) {
+    __updateSorter(sorter, shiftClick) {
       if (!sorter.direction && this._sorters.indexOf(sorter) === -1) {
         return;
       }
 
       sorter._order = null;
 
-      if (this.multiSort) {
+      if (this.multiSort || (this.multiSortOnShiftClick && shiftClick)) {
         if (this.multiSortPriority === 'append') {
           this.__appendSorter(sorter);
         } else {
           this.__prependSorter(sorter);
         }
-      } else if (sorter.direction) {
+      } else if (sorter.direction || this.multiSortOnShiftClick) {
         const otherSorters = this._sorters.filter((s) => s !== sorter);
-        this._sorters = [sorter];
+        this._sorters = sorter.direction ? [sorter] : [];
```

Comment:
Seems to be missing a test case

Reply:
This line was added for [this](https://github.com/vaadin/web-components/blob/4a3a50ab20fb72b7906d64fdd2986bb2c2714990/packages/grid/test/sorting.test.js#L274-L278) test case to pass

Reply:
Hmm...I get all green test run even if I revert this line to its original form:

<img width="695" alt="Screenshot 2022-11-01 at 15 17 43" src="https://user-images.githubusercontent.com/1222264/199242308-c5205545-0c9c-46f4-8c73-f06233b4a5e0.png">

Reply:
Ah ok I see. Mb, will add more test cases

Reply:
Done.